### DISCUSSION TITLE
Proposal: Confidence Field

### DISCUSSION THREAD

**Discussion by CTM1** [SigmaHQ/sigma-specification#153] [Ideas]
When providing Sigma rules to user, some providers seem to use "Level" interchangeably with what would be "Confidence".

[The doc](https://sigmahq.io/docs/basics/rules.html#metadata-leve) says `The level field describes the criticality of a triggered rule.`  and `critical should never trigger a false positive and be of high relevance`. The presence/absence of false positives and the relevance/criticality of an alert should be two different metrics to me.

Take for instance this Process Hollowing rule.

https://github.com/SigmaHQ/sigma/blob/master/rules/windows/process_tampering/proc_tampering_susp_process_hollowing.yml

This for me would be a `high/critical` level with `medium` confidence, but its level is now set to `medium` as many products (AV, Games etc.) modify their image in memory. But in itself, if this event is not a false-positive, it would be a high/critical alert.

Adding this field would allow for better modular configuration of  rule sets.

I would volunteer to make a PR for this field myself, if it's something that seems relevant to the community.

EDIT: Confidence should be an optional field

**Comment by Neo23x0** [comment]
Could you please explain how a rule with "level: critical + confidence: medium" would be treated differently (before a match) than a rule with "level: medium" today?

**Reply by CTM1** [reply]
First let me define `level` and `confidence`:

`level`: Indicates the criticality or severity of a triggered rule, i.e. the potential impact or importance of the detected event.

`confidence`: The degree of certainty that a triggered rule represents a true positive. The reliability of the rule, much like the field used in IOCs today. [[OpenCTI]](https://docs.opencti.io/latest/usage/reliability-confidence/) [[STIX]](https://docs.oasis-open.org/cti/stix/v2.1/csprd01/stix-v2.1-csprd01.html#_Toc16070848)

From what I understand "before a match":


- An end-user perspective could be: 

"I don't want many false positives", today this would be hard to do, he may decide to activate critical rules only, but he'd lose precious information in high rules that don't cause many false positives.  If there was this new field, he could activate only high confidence rules and keep a relative low false positive rate while covering a wider range of events. This would allow users to progressively whitelist rules when building their SOC.

- A vendor perspective could be: 

"I have two rules, 

One that raises many FPs and could be quite critical (e.g. https://github.com/SigmaHQ/sigma/blob/master/rules/windows/process_access/proc_access_win_lsass_memdump.yml)

One that doesn't raise many FPs but isn't as critical (e.g. https://github.com/SigmaHQ/sigma/blob/master/rules/windows/registry/registry_set/registry_set_change_rdp_port.yml)"

They're both tagged `high`, if I want to dissociate them, (for end-user distribution or internal use), I have no metric through which to do so, either change RDP port change to "critical" or the LSASS "memdump" to medium, which I believe we'd agree wouldn't make sense. Adding confidence would allow to distinguish between them.

This is sort of taken into account with certain vendors. [Sekoia](https://docs.sekoia.io/xdr/features/detect/rules_catalog/#effort-level), [Trend-Micro](https://www.trendmicro.com/vinfo/us/security/definition/confidence-level)

"After a match", this could be used to prioritize investigation for SoC teams and decide which rules to  "block" for EDR vendors.

**TL;DR**
To answer your question directly,  
- An analyst might skip over "medium" rules but not "critical" rules even if he has a high false positive tolerance. This would provide nuance to investigate those alerts.
- A vendor might recommend to a client with low false positive tolerance to only activate high confidence rules in a first run, and progressively whitelist the lower confidence rules as they activate them while providing better coverage.

**Reply by Neo23x0** [reply]
From my perspective, adding a "confidence" field to the Sigma rule standard raises one major and a few minor concerns.

Currently, when setting the level of a rule, we primarily evaluate its relevance by asking, "How significant is a match with this rule?" While the confidence we have about potential false positives does influence the level, it plays a relatively minor role. This is because, as detection engineers, we often lack the necessary visibility and telemetry to accurately assess confidence levels.

The example you mentioned is one of the rare cases where confidence can be reasonably estimated, as you’ve shown. However, only MSSPs or SIEM service providers have the advantage of working with a broad and diverse dataset from multiple sources and tenants. This gives them the ability to set confidence levels with the certainty I would expect for such a field. If we're forced to guess 90% of the time, and those guesses frequently miss the mark, I'd rather omit a "confidence" field altogether than risk adding misleading information.

**Reply by CTM1** [reply]
I failed to communicate this, but for me confidence would be an optional field, as [level already is](https://github.com/SigmaHQ/sigma-specification/blob/main/specification/sigma-rules-specification.md#level), so its usage wouldn't be forced onto people who don't have the capabilities to assess it.

For me adding this field would simply allow MSSPs/SIEM/whoever to stay compliant with this widely used standard if they so wish to add this granularity, not enforce a review of all current public SigmaHQ rules. I used SigmaHQ's rules as an example because those are public rules which could illustrate my point, but it's something contributors to SigmaHQ who don't have the resources to assess it won't add and it could be excluded from rules in which they wouldn't make sense. (But would allow contributors that do to add it with sufficient proof, I suppose).

Does this address your concern/raise others ?

**Reply by Neo23x0** [reply]
So far, in the SigmaHQ rule set, we've only added optional fields (such as level, falsepositives, and date) when at least 95% of the rules could be populated with them. We’ve avoided adding optional fields to only a small subset of rules to prevent user confusion and maintain consistency across the field selection for all rules.

We should bring in @nasbench to get his opinion on this.

**Reply by CTM1** [reply]
I understand the desire for consistency in the SigmaHQ ruleset. However, I think we need to consider Sigma more broadly as an open standard, not just in the context of the SigmaHQ ruleset.

While SigmaHQ maintains an excellent public ruleset, it's just one implementation of Sigma rules. Many organizations and vendors create their own private Sigma rulesets tailored to their specific needs and capabilities.

Adding an optional confidence field to the Sigma specification simply provides the capability for those who find it valuable. Organizations with the telemetry and expertise to accurately assess confidence levels could leverage this field in their own private rulesets or by enriching the public ones.

I see what you mean when you say "could be populated", but most optional fields, like "scope" aren't used in 95% of rules, so I fail to see the practical aspect of this in terms of consistency or user confusion.

Separate from its immediate application in any particular ruleset, I still believe it would be a positive addition to the specification. I do we believe we could benefit from second opinions from other maintainers :+1: .

**Reply by Neo23x0** [reply]
> Many organizations and vendors create their own private Sigma rulesets tailored to their specific needs and capabilities.
> Adding an optional confidence field to the Sigma specification simply provides the capability for those who find it valuable.

They can already do that. 

> Organizations with the telemetry and expertise to accurately assess confidence levels could leverage this field in their own private rulesets or by enriching the public ones.

I don't think many organisations can do that. 
MSSPs, yes. 
But even a single big corporation has standards, e.g. they use SAP instead of Peoplesoft as the ERP and cannot know it a certain rule triggers false positives caused by Peoplesoft applications. 

>  but most optional fields, like "scope" aren't used in 95% of rules, so I fail to see the practical aspect of this in terms of consistency or user confusion.

I'm not a fan of the `scope` field. I'd like to remove it and any other field that isn't used in the majority of the the rules from the specification. It's a legacy issue and because others wanted to include it.
